grudge/reductions.py

```diff
@@ -289,7 +300,9 @@ def _apply_elementwise_reduction(
     @memoize_in(actx, (_apply_elementwise_reduction,
                        "elementwise_%s_prg" % op_name))
     def elementwise_prg():
-        return make_loopy_program(
+        # FIXME: This computes the reduction value redundantly for each
+        # output DOF.
```

Comment:
This looks eerily similar to the same reduction in pytential, so probably needs fixing there too.

Reply:
https://github.com/inducer/pytential/pull/100